pkg/providers/tinkerbell/hardware/config.go

```diff
@@ -80,6 +81,10 @@ func (hc *HardwareConfig) ValidateHardware() error {
 			return fmt.Errorf("hardware: %s ID is required", hw.Name)
 		}
 
+		if _, ok := hardwares[hw.Spec.ID]; !ok {
```

Comment:
Why did we opt to move this here? I don't think the error makes sense now because it infers stuff about being registered with the tinkerbell stack but does do anything with the tinkerbell stack.